cmake/compiler/xt-clang/target.cmake

```diff
@@ -16,7 +14,20 @@ set(LLEXT_REMOVE_FLAGS
 
 # Flags to be added to llext code compilation
 set(LLEXT_APPEND_FLAGS
-  -fPIC
   -nostdlib
   -nodefaultlibs
 )
+
+if(CONFIG_LLEXT_BUILD_PIC)
+set(LLEXT_REMOVE_FLAGS ${LLEXT_REMOVE_FLAGS}
+  -fno-pic
+  -fno-pie
+)
+set(LLEXT_APPEND_FLAGS ${LLEXT_APPEND_FLAGS}
+  -fPIC
+)
+else()
+set(LLEXT_APPEND_FLAGS ${LLEXT_APPEND_FLAGS}
+  -ffreestanding
```

Comment:
Interesting - never saw `-ffreestanding` before, so I have read about it... IIUC it makes the llext use the core's definition of the standard functions. Shouldn't this be the default for both PIC and non-PIC? (... and on other arches as well?)

Also, should we add `-fno-pic` here? Not sure what the Zephyr default is :man_shrugging:

[whatever the decision, also update `gcc/target_xtensa.cmake` to match]